### PR TITLE
[#10164] Result Page: InstructorSessionResultSectionType.GIVER still shows recipient name

### LINKED ISSUE #10164: Result Page: InstructorSessionResultSectionType.GIVER still shows recipient name
- **Environment**: master

**Steps to reproduce**
- Go to result page for "Session with different question types"
- Select options as below
image
- Click "Tutorial Group 1"

**Expected behaviour**
Only shows giver from Tutorial Group 1

**Actual behaviour**
Participant from Tutorial Group 2 is shown, though there is no response (which is expected)
image

**Additional info**
The same issue exist for GQR/RGQ also I belive.

### PR DESCRIPTION
**Pull request by t-cheepeng**
Fixes #10164 

**Outline of Solution**
**GRQ**
gqr

**GQR**
f

The bug was due to the templates for the components not checking against the responses to show. The fix was to add a `ngIf` check
- Also added a new variable to both views so that the `Team xxx` tab/card will not show if there are no responses from anyone in the team

### CODE REVIEW

**Comment by xpdavid** [comment]
Good attempt. However, this is not the root cause of the bug. Try to figure out why the tab is there in the first place (Your current fix just assumes the tab is there and try to hide it if there is no response).

**Comment by t-cheepeng** [comment]
> 
> 
> Good attempt. However, this is not the root cause of the bug. Try to figure out why the tab is there in the first place (Your current fix just assumes the tab is there and try to hide it if there is no response).

Is the root cause in the `teamsToUsers` record? In the filtering of responses, the names that do not belong to the section are not removed, making them be rendered

**Comment by xpdavid** [comment]
Yup!

File: src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.ts [hunk 1]
```diff
@@ -142,6 +142,12 @@ export class GqrRqgViewResponsesComponent extends InstructorResponsesViewBase im
             .isFeedbackResponsesDisplayedOnSection(response, this.section, this.sectionType);
 
           if (!shouldDisplayBasedOnSection) {
+            const team: string = this.isGqr ? response.giverTeam : response.recipientTeam;
```
Comment (xpdavid) [inline src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.ts:145]: Have you considered put the same `shouldDisplayBasedOnSection` check in L89?

File: src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.ts [hunk 2]
```diff
@@ -92,33 +92,39 @@ export class GqrRqgViewResponsesComponent extends InstructorResponsesViewBase im
...
-          if (!response.recipientTeam) {
-            // Recipient is team
-            this.teamsToUsers[response.recipient] = this.teamsToUsers[response.recipient] || [];
-            if (this.teamsToUsers[response.recipient].indexOf(response.recipient) === -1) {
-              this.teamsToUsers[response.recipient].push(response.recipient);
-              this.teamExpanded[response.recipient] = this.isExpandAll;
+
+        const shouldDisplayBasedOnSection: boolean = this.feedbackResponsesService
+            .isFeedbackResponsesDisplayedOnSection(response, this.section, this.sectionType);
+        if (shouldDisplayBasedOnSection) {
```
Comment (xpdavid) [inline src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.ts:98]: Try to use guard clause?

```java
const shouldDisplayBasedOnSection = xx

if (!shouldDisplayBasedOnSection) {
    continue;
}

```

**Pull request merged**